torch/_refs/__init__.py

```diff
@@ -1276,8 +1276,17 @@ def isclose(
 
 
 def _lcm(a: TensorLikeType, b: TensorLikeType):
-    g = gcd(a, b)
-    return where(eq(g, 0), 0, abs(mul(true_divide(a, g), b)))
+    dtype = a.dtype
+    promote_to_int = dtype in (torch.int8, torch.int16)
+    if promote_to_int:
+        a = prims.convert_element_type(a, torch.int32)
+        b = prims.convert_element_type(b, torch.int32)
+
+    g = torch.gcd(a, b)
+    # Avoid division by zero in case gcd(0, 0) == 0
+    g = torch.where(g == 0, 1, g)
+    res = torch.abs(prims.div(a, g) * b)
```

Comment:
is it possible for `a` and `g` to have different types (e.g. a is int32, and b is int64, which probably will result in g int64). `prim.div` won't work for this case?

Reply:
Both operators call `_make_elementwise_binary_reference` with the same `type_promotion_kind` so I don't think that should ever happen.